Gems/ROS2/Code/Source/FactorySimulation/ConveyorBeltComponentKinematic.cpp

```diff
@@ -0,0 +1,306 @@
+        }
+        return AZ::Vector3::CreateZero();
+    }
+
+    AZStd::pair<float, AzPhysics::SimulatedBodyHandle> ConveyorBeltComponentKinematic::CreateSegment(
+        AZ::ConstSplinePtr splinePtr,
+        AzPhysics::SystemInterface* physicsSystem,
+        AzPhysics::SceneHandle sceneHandle,
+        float normalizedLocation)
+    {
+        auto colliderConfiguration = AZStd::make_shared<Physics::ColliderConfiguration>();
+        colliderConfiguration->m_materialSlots.SetMaterialAsset(0, m_materialAsset);
+        auto shapeConfiguration =
+            AZStd::make_shared<Physics::BoxShapeConfiguration>(AZ::Vector3(m_segmentLength, m_beltWidth, m_segmentWidth));
+        const auto transform = GetTransformFromSpline(m_splineConsPtr, normalizedLocation);
+        AzPhysics::RigidBodyConfiguration conveyorSegmentRigidBodyConfig;
+        conveyorSegmentRigidBodyConfig.m_kinematic = true;
+        conveyorSegmentRigidBodyConfig.m_position = transform.GetTranslation();
+        conveyorSegmentRigidBodyConfig.m_orientation = transform.GetRotation();
+        conveyorSegmentRigidBodyConfig.m_colliderAndShapeData = AzPhysics::ShapeColliderPair(colliderConfiguration, shapeConfiguration);
+        conveyorSegmentRigidBodyConfig.m_computeCenterOfMass = true;
+        conveyorSegmentRigidBodyConfig.m_computeInertiaTensor = true;
+        conveyorSegmentRigidBodyConfig.m_startSimulationEnabled = true;
+        conveyorSegmentRigidBodyConfig.m_computeMass = false;
+        conveyorSegmentRigidBodyConfig.m_mass = 1.0f;
+        conveyorSegmentRigidBodyConfig.m_entityId = GetEntityId();
+        conveyorSegmentRigidBodyConfig.m_debugName = "ConveyorBeltSegment";
+        AzPhysics::SimulatedBodyHandle handle = physicsSystem->GetScene(sceneHandle)->AddSimulatedBody(&conveyorSegmentRigidBodyConfig);
+        AZ_Assert(handle == AzPhysics::InvalidSimulatedBodyHandle, "Body created with invalid handle");
+        return AZStd::make_pair(normalizedLocation, handle);
+    }
+
+    void ConveyorBeltComponentKinematic::OnTick(float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        AzPhysics::SystemInterface* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get();
+        AZ_Assert(physicsSystem, "No physics system");
+        AzPhysics::SceneInterface* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AZ_Assert(sceneInterface, "No scene interface");
+        AzPhysics::SceneHandle defaultSceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+        AZ_Assert(defaultSceneHandle != AzPhysics::InvalidSceneHandle, "Invalid default physics scene handle");
+
+        // Initialization. Create segments and add to scene, attach post-simulation event, cache pointers,transform etc.
+        // Strong assumption is that conveyor belt is not transformed after initialization.
+        if (!m_initialized)
+        {
+            m_splineTransform = AZ::Transform::CreateIdentity();
```

Comment:
This could be moved to subroutine to help with understanding what is happening in onTick